compute_tools/src/compute.rs

```diff
@@ -211,6 +211,44 @@ pub struct ParsedSpec {
     pub endpoint_storage_token: Option<String>,
 }
 
+impl ParsedSpec {
+    pub fn validate(&self) -> Result<&Self, String> {
+        // Only Primary nodes are using safekeeper_connstrings, and at the moment
+        // this method only validates that part of the specs.
+        if self.spec.mode != ComputeMode::Primary {
+            return Ok(self);
+        }
+
+        // While it seems like a good idea to check for an odd number of entries in
+        // the safekeepers connection string, changes to the list of safekeepers might
+        // incur appending a new server to a list of 3, in which case a list of 4
+        // entries is okay in production.
+        //
+        // Still we want unique entries, and at least one entry in the vector
+        if self.safekeeper_connstrings.is_empty() {
+            return Err(String::from("safekeeper_connstrings is empty"));
+        }
+
+        // check for unicity of the connection strings
+        let mut connstrings = self.safekeeper_connstrings.clone();
+
+        connstrings.sort();
+        let previous = &connstrings[0];
+
+        for current in connstrings.iter().skip(1) {
+            // duplicate entry?
+            if current == previous {
```

Comment:
I don't see `previous` updated anywhere in the loop

Reply:
Oh my. Thanks. Update coming soon.

Reply:
Updated, with a Rust unit test exercising that parts of the code.